### PR TITLE
fix: Changed slide in/out button

### LINKED ISSUE #6131: Virtual Events: Implement a different slide-in/out button for video sidebar
Please implement a slide button in a more similar way to Gmail in style and position (next to screen edge), but make the slider button square with rounded corners. Also change the X closing button to an inverse < slider button.

Compare
Screenshot from 2020-12-24 10-34-22

Screenshot from 2020-12-08 15-05-35

### PR DESCRIPTION
**Pull request by nlok5923**
Fixes #6131 

#### Short description of what this resolves:
implements a slide in/out button

#### Changes proposed in this pull request:

- Changed the slide in/out button

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

Screenshot of changes :
image
image

### CODE REVIEW

**Comment by mariobehling** [comment]
Please change the following:
* Keep button in the top. Do not move it to the middle of the page.
* Do not make the corners rounded to the edge of the screen.

**Comment by nlok5923** [comment]
@mariobehling  sir i had make the  changes but sir the semantic ui button are default with rounded corners.

**Comment by nlok5923** [comment]
changes in latest commit.
image

**Comment by iamareebjamal** [comment]
`left: -5px`

**Comment by nlok5923** [comment]
changes in latest commit:
image

**Comment by mariobehling** [comment]
Please reduce the button width a bit.

**Comment by afzal442** [comment]
Hi, I think you didn't observe Gmail in style and position. You have to replace the icon. Thanks.

**Comment by mariobehling** [comment]
It is ok. Keep it there for now. We will test if in practice a few days.

But please also make the change of the button on the right hand side of the sidebar.

Screenshot from 2020-12-24 13-51-26

**Comment by nlok5923** [comment]
changes in latest commit:
image
changed X to <
image

**Comment by afzal442** [comment]
Hi, I think you again forgot to change the button on the right side of the sidebar as per above suggestion. Plz have a look at screenshot above and make changes.

**Comment by nlok5923** [comment]
@afzal442 please can you tell me the step to reproduce the same screen shot as I only find the cross X icon on the right side of sidebar.

**Comment by afzal442** [comment]
> the same screen shot as I only find the cross X icon on the right side of sidebar.

Yours
c3
Expected
c2

**Comment by nlok5923** [comment]
Thanks @afzal442 will do that changes.

**Comment by nlok5923** [comment]
changes in latest commit :
image

File: app/components/public/stream/side-panel.hbs [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
-  <i class="arrow right icon"></i>
+<button {{did-insert this.setup}} class="ui icon small button" style="position: absolute; top: 0; margin-top: 10px; left: -5px" {{action (mut this.shown) true}}>
+  <i class="angle right icon"></i>
 </button>
 {{#if this.shown}}
   <div class="d-flex flex-column p-4"
     style="width: 300px; height: 100%; position: absolute; top: 0; background-color: #555; color: white;" >
-    <i role="button" class="close icon ml-auto" {{action (mut this.shown) false}}></i>
+    <button class="ui icon small button ml-auto" {{action (mut this.shown) false}}>  
+    <i class="angle left icon"></i>
```
Comment (daretobedifferent18) [inline app/components/public/stream/side-panel.hbs:8]:
```suggestion
      <i class="angle left icon"></i>
```

File: app/components/public/stream/side-panel.hbs [hunk 2]
```diff
@@ -1,10 +1,12 @@
-<button {{did-insert this.setup}} class="ui icon button" style="position: absolute; top: 0; margin: 10px" {{action (mut this.shown) true}}>
-  <i class="arrow right icon"></i>
+<button {{did-insert this.setup}} class="ui icon small button" style="position: absolute; top: 0; margin-top: 10px; left: -5px" {{action (mut this.shown) true}}>
```
Comment (daretobedifferent18) [inline app/components/public/stream/side-panel.hbs:1]: instead on styling with inline style, make a class and define styling there

**Comment by mariobehling** [comment]
> Please reduce the button width a bit.

File: app/components/public/stream/side-panel.hbs [hunk 3]
```diff
@@ -1,10 +1,12 @@
-<button {{did-insert this.setup}} class="ui icon button" style="position: absolute; top: 0; margin: 10px" {{action (mut this.shown) true}}>
-  <i class="arrow right icon"></i>
+<button {{did-insert this.setup}} class="ui icon small button toggle-btn" style="position: absolute; top: 0; margin-top: 10px; left: -5px" {{action (mut this.shown) true}}>
```
Comment (maze-runnar) [inline app/components/public/stream/side-panel.hbs:1]: move styling to toggle-btn class

Comment (nlok5923) [reply app/components/public/stream/side-panel.hbs:1]: toggle btn has it's own defination of styling moving inline stlying to toggle btn class will break the styling for the right side button of side panel.

Comment (maze-runnar) [reply app/components/public/stream/side-panel.hbs:1]: have you tried it? try once, if it didn't work create new class move style there.

**Comment by nlok5923** [comment]
changes after latest commit ;
image
image

**Comment by nlok5923** [comment]
@mariobehling  @iamareebjamal sir please review this pull request.

**Comment by maze-runnar** [comment]
bring that icon in middle, everything else is good.
Screenshot from 2020-12-29 08-23-19

**Comment by iamareebjamal** [comment]
Also, the icon should be stuck to the right edge of the side panel, there is a margin there currently

**Comment by nlok5923** [comment]
Screenshot after latest commit:
image

**Comment by nlok5923** [comment]
@iamareebjamal sir please review the changes.

**Comment by iamareebjamal** [comment]
Compare image

It should look the same but mirrored. Read the comments above. The initial comment was to remove rounded corners from the edge side. You need to review things yourself first and then request the mentors. Design needs to be symmetric and consistent

File: app/components/public/stream/side-panel.hbs [hunk 4]
```diff
@@ -1,10 +1,14 @@
...
-  <i class="arrow right icon"></i>
+<button {{did-insert this.setup}} class="btn-position ui icon small button toggle-btn" {{action (mut this.shown) true}}>
+  <i class="angle right icon"></i>
 </button>
 {{#if this.shown}}
   <div class="d-flex flex-column p-4"
     style="width: 300px; height: 100%; position: absolute; top: 0; background-color: #555; color: white;" >
-    <i role="button" class="close icon ml-auto" {{action (mut this.shown) false}}></i>
+    <div class="ml-auto set-margin" >
+    <button class="ui icon small button toggle-btn ml-auto" {{action (mut this.shown) false}}>
```
Comment (afzal442) [inline app/components/public/stream/side-panel.hbs:8]: use rounded-none

**Review by afzal442 — CHANGES_REQUESTED** [review]

**Comment by iamareebjamal** [comment]
@nlok5923 Please complete the PR or close it

**Comment by nlok5923** [comment]
@iamareebjamal please can you provide me a clue of how can i remove the border of button as sir those are semantic ui btn .

**Comment by iamareebjamal** [comment]
border css style lets you define all four border radiuses

**Comment by nlok5923** [comment]
Sir class based styling do not allow us to modify the default semantic UI styling I need to use id to modify the default semantic UI button style shall I use it sir ?.

**Comment by iamareebjamal** [comment]
> Sir class based styling do not allow us to modify the default semantic UI styling

Impossible

**Comment by iamareebjamal** [comment]
You have overriden my changes from upstream while resolving conflicts. I have to recreate the PR now

**Comment by nlok5923** [comment]
Okay I will close this pr and will create a new pr for resolving this issue

**Comment by iamareebjamal** [comment]
image
image

File: app/components/public/stream/side-panel.hbs [hunk 5]
```diff
@@ -1,10 +1,12 @@
...
+<button {{did-insert this.setup}} {{did-update this.reOrderStreams @videoStream}} class="ui icon button stream-side-panel-button" {{action (mut this.shown) true}}>
+  <i class="angle right icon"></i>
 </button>
 {{#if this.shown}}
-  <div class="d-flex flex-column p-4 stream-side-panel">
-    <i role="button" class="close icon ml-auto" {{action (mut this.shown) false}}></i>
-
+  <div class="d-flex flex-column p-4 stream-side-panel" >
+    <button class="ui icon button stream-side-panel-button right" {{action (mut this.shown) false}}>  
+     <i class="angle left icon"></i>
```
Comment (iamareebjamal) [inline app/components/public/stream/side-panel.hbs:7]:
```suggestion
      <i class="angle left icon"></i>
```

**Pull request merged**